### PR TITLE
Write to log file and add `--upload-metadata` to publish metadata files

### PR DESCRIPTION
**Pull request by ghengeveld**
Fixes AP-3531, AP-2517

First of all, this updates our logging to write the CLI logs to a `chromatic.log` file besides printing to the terminal. The log file format is akin to what the logs look like in CI.

Secondly, this adds an option to upload a set of "metadata" files, which will be available as part of the published Storybook in the `.chromatic` directory. A [directory index](https://5d67dc0374b2e300209c41e7-nyshnotsif.chromatic.com/.chromatic/) is provided:

Screenshot 2023-10-12 at 12 57 45

<details>
  <summary>📦 Published PR as canary version: <code>7.5.0--canary.836.6619467092.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@7.5.0--canary.836.6619467092.0
  # or 
  yarn add chromatic@7.5.0--canary.836.6619467092.0
  ```
</details>

### CODE REVIEW

**Review by tevanoff — COMMENTED** [review]
Love this, this is great! However, you say:

> these files will be public on a public Storybook.

That sounds like a deal breaker to me. Can we sync up on options for storing these more securely?

**Comment by tmeasday** [comment]
@ghengeveld perhaps change the PR description as this could unduly worry someone who reads this PR.

File: node-src/index.ts [hunk 1]
```diff
@@ -102,17 +109,26 @@ export async function run({
...
+
   // Run these in parallel; neither should ever reject
   await Promise.all([runBuild(ctx), checkForUpdates(ctx)]);
 
   if (ctx.exitCode === 0 || ctx.exitCode === 1) {
     await checkPackageJson(ctx);
   }
 
-  if (ctx.options.diagnostics) {
+  if (ctx.flags?.diagnostics || ctx.extraOptions?.diagnostics) {
```
Comment (tmeasday) [inline node-src/index.ts:125]: `runBuild` should have set `ctx.options` from `flags`/`extraOptions`, so this change is redundant. But I don't love the indirection.

**Review by tmeasday — APPROVED** [review]
LGTM, good stuff

File: node-src/index.ts (hunk 1 above)
Comment (ghengeveld) [reply node-src/index.ts:125]: `runBuild` may throw before it even sets `ctx.options` (e.g. when parsing a configuration file or flags). This is further addressed in [this PR](https://github.com/chromaui/chromatic-cli/pull/840).

**Pull request merged**